pkg/plugins/policies/meshhttproute/xds/configurer.go

```diff
@@ -22,22 +24,34 @@ func (c RoutesConfigurer) Configure(virtualHost *envoy_route.VirtualHost) error
 	matches := c.routeMatch(c.Matches)
 
 	for _, match := range matches {
-		r := &envoy_route.Route{
-			Match: match.routeMatch,
-			Action: &envoy_route.Route_Route{
+		rb := &route.RouteBuilder{}
+
+		rb.Configure(route.RouteMustConfigureFunc(func(envoyRoute *envoy_route.Route) {
+			// todo: create configurers for Match and Action
+			envoyRoute.Match = match.routeMatch
+			envoyRoute.Action = &envoy_route.Route_Route{
 				Route: c.routeAction(c.Split),
-			},
-			TypedPerFilterConfig: map[string]*anypb.Any{},
-		}
+			}
+			envoyRoute.TypedPerFilterConfig = map[string]*anypb.Any{}
+		}))
 
 		// We pass the information about whether this match was created from
 		// a prefix match along to the filters because it's no longer
 		// possible to know for sure with just an envoy_route.Route
 		for _, filter := range c.Filters {
-			routeFilter(filter, r, match.prefixMatch)
+			rb.
+				Configure(filters.NewRequestHeaderModifier(filter.RequestHeaderModifier)).
```

Comment:
I'd rather have a `FilterConfigurer` with a switch statement. The way it is now obfuscates the fact that each filter is exactly one of these and when accessing the type we must _always_ check the `filter.Type` before trying to get access to a variant.

Reply:
Why do we need `filter.Type`? It's not how we usually do `oneof` in Kuma, right?. For example backend `MeshTrace` doesn't have `backend.Type`:
```go
// Only one of zipkin, datadog or openTelemetry can be used.
type Backend struct {
	// Zipkin backend configuration.
	Zipkin *ZipkinBackend `json:"zipkin,omitempty"`
	// Datadog backend configuration.
	Datadog *DatadogBackend `json:"datadog,omitempty"`
	// OpenTelemetry backend configuration.
	OpenTelemetry *OpenTelemetryBackend `json:"openTelemetry,omitempty"`
}
```

Reply:
Well the immediate reason is that it's the Gateway API style but that is only because it's [how Kubernetes has decided to do unions in APIs](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md#discriminator-field), see [their reasoning](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md#non-discrminated).

Reply:
I would be happy with just a `FilterConfigurer` as opposed to checking all of the possible variants inline. It doesn't _need_ to have a `switch`, I suppose, though IMO that leads to more readable code because it's clear when you check `type` that it's really _exactly one_ of these filter types.

Reply:
Do we want to update other Kuma policies in that case?

Reply:
IMO not a bad idea, at least for new policies. Maybe @lahabana has an opinion here.

Reply:
Can discuss here https://github.com/kumahq/kuma/issues/6069